controller/googleOauthHandler.js

```diff
@@ -0,0 +1,102 @@
+require('dotenv').config({
+    path: '../config/variables.env'
+});
+
+const googleOAuth = require('../middleware/googleOAuthClient');
+const User = require('../models/user');
+const tokenGenerator = require('./accessTokenGenerator');
+
+
+
+module.exports.getMessage = async (googleToken) => {
+    try {
+        const response = await googleOAuth.getUserInfo(googleToken);
+        const userInfo = response.data;
+
+        const token = await tokenGenerator.token(userInfo.email);
+        const refresh_token = await tokenGenerator.refresh_token(userInfo.email);
+
+        try {
+            //If the google id already exists in the db, store nothing to the db
+            if (await dbQuery.idExists(userInfo)) {
+                return { 
+                    message: token,
+                    statusCode: 200
+                };
+            } else {
+                //if the google email exists in the db but is not linked, link the accoints
```

Comment:
comments are something you want the other person to read, it would be greatly appreciated if you checked for spelling errors,